cns-hook-util

```diff
@@ -71,51 +71,84 @@ function update_remotevm {
 		sdc-vmadm update $vmuuid
 }
 
+function find_path {
+    local path="$1"
+    if [ ! -e "$path" ]; then
+        path=/native/$1
+        if [ ! -e "$path" ]; then
+            echo "ERROR: failed to find $(basename $path) command" >&2
+            echo "       is this a Triton zone?" >&2
+            exit 1
+        fi
+    fi
+    cmd="$path"
+    if [ -e "/usr/bin/pfexec" ]; then
+        cmd="/usr/bin/pfexec ${cmd}"
+    fi
+    echo "$cmd"
+}
+
+function mdata_get {
+    local cmd=$(find_path /usr/sbin/mdata-get)
+    $cmd "$@"
+}
+
 function mdata_put {
-	local path=/usr/sbin/mdata-put
-	if [ ! -e "$path" ]; then
-		path=/native/usr/sbin/mdata-put
-		if [ ! -e "$path" ]; then
-			echo "ERROR: failed to find mdata-put command" >&2
-			echo "       is this a Triton zone?" >&2
-			exit 1
-		fi
-	fi
-	cmd="$path"
-	if [ -e "/usr/bin/pfexec" ]; then
-		cmd="/usr/bin/pfexec ${cmd}"
-	fi
-	$cmd "$@"
+    local cmd=$(find_path /usr/sbin/mdata-put)
+    $cmd "$@"
 }
 
 function mdata_delete {
-	local path=/usr/sbin/mdata-delete
-	if [ ! -e "$path" ]; then
-		path=/native/usr/sbin/mdata-delete
-		if [ ! -e "$path" ]; then
-			echo "ERROR: failed to find mdata-delete command" >&2
-			echo "       is this a Triton zone?" >&2
-			exit 1
-		fi
+    local cmd=$(find_path /usr/sbin/mdata-delete)
+	$cmd "$@"
+}
+
+function merge_token {
+	local domain="${1}" tokenval="${2}"
+	local existing_token
+
+	local zname=$(zonename)
```

Comment:
Should we make all the `$(zonename)` into a `function zonename` like the `mdata_*` ones? I thought you pasted a log of it failing on LX from not being able to find the zonename command?